components/inputs/test/input-text.visual-diff.js

```diff
@@ -0,0 +1,53 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-input-text', function() {
+
+	const visualDiff = new VisualDiff('input-text', __dirname);
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 800, height: 800, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/inputs/test/input-text.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
+	[
+		'basic',
+		'email',
+		'number',
+		'password',
+		'search',
+		'tel',
+		'url',
+		'disabled',
+		'placeholder',
+		'placeholder-disabled',
+		'invalid',
+		'invalid-disabled',
+		'aria-invalid',
+		'aria-invalid-disabled',
+		'override-height',
+		'override-padding',
+		'override-placeholder'
+	].forEach((name) => {
+		it(name, async function() {
+			const rect = await visualDiff.getRect(page, `#${name}`);
+			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+		});
+		it(`${name}-focus`, async function() {
+			await page.evaluate((name) => {
+				const elem = document.querySelector(`#${name}`);
+				elem.focus();
+			}, name);
+			const rect = await visualDiff.getRect(page, `#${name}`);
```

Comment:
I think we could run into inconsistency in the screenshots resulting from the `background-color` transition.  Speeding up the playback of the animation like I did for dialogs might be sufficient, but waiting for the animations to complete would also be safe(r).

Reply:
Note however, I think `transitionend` is not ` composed`, so need to be careful what to listen on.  Sometimes I have found it nice to have a helper like:

```
const getTransitionEvent = (page, selector) => {
    return page.$eval(selector, (elem) => {
        return new Promise((resolve) => {
            elem.shadowRoot.addEventListener('transitionend', resolve, { once: true });
        });
    });
}
```
and then the test can be fairly simple still...
```
const transitionEvent = getTransitionEvent(page, `#${name}`);
await page.$eval(`#${name}`, (elem) => elem.focus());
await transitionEvent;
await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
```
Also note that `transitionend` will get dispatched twice.  Once for the border (the super fast one), and a second time for the background.  The property being transitioned is passed to event handlers, so it is possible to check that if necessary.  I think `d2l-button-icon` awaits for both `box-shadow` and `background-color`.

Reply:
Both transitions are now removed, so should be a non-issue now. But good to know for future!